.github/workflows/on-pullrequest-powershell.yml

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: On Pull Request - PowerShell
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**/*.ps1'
+      - '**/*.psm1'
+      - '**/analyze-repository.yml'
+      - '**/powershell-analyzer.yml'
+  workflow_dispatch:
+
+jobs:
+  run-lint:
+    name: Linter
+    uses: Ed-Fi-Alliance-OSS/Ed-Fi-Actions/.github/workflows/powershell-analyzer.yml@main
+
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the Actions repo
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846  # v3.0.0
+
+      - name: Run Unit Tests
+        shell: pwsh
+        working-directory: powershell-analyzer
+        run: |
+          $PesterConfig = New-PesterConfiguration
+          $PesterConfig.TestResult.OutputFormat = "JUnitXml"
+          $PesterConfig.TestResult.OutputPath = "ps-analyzer.xml"
+          $PesterConfig.TestResult.Enabled = $True
+          Invoke-Pester -Configuration $PesterConfig
+
+      - name: Upload Test Results
+        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535 # v3.0.0
+        with:
+          name: powershell-analyzer-test-results
+          path: powershell-analyzer/ps-analyzer.xml
+          retention-days: 5
```

Comment:
I took a break from the SBOM and provenance work to learn about PowerShell unit testing, and thereby move toward a higher repository score.